nb-engine/nb-engine-core/src/main/java/io/nosqlbench/engine/api/activityimpl/uniform/Activity.java

```diff
@@ -0,0 +1,969 @@
+/// Activities and their containers are hosted within an [NBSession]. Activities run asynchronously
+/// with respect to their container and session. There are a few session commands which
+/// are used to manage activities, such as [start][CMD_start], [stop][CMD_stop], and
+/// [await][CMD_await], for example.
+///
+/// The config parameters for an activity are standard, and custom behaviors afforded to activities
+/// work the same across all op types.
+public class Activity<R extends java.util.function.LongFunction, S> extends NBStatusComponent
+    implements InvokableResult,
+               SyntheticOpTemplateProvider,
+               StateCapable,
+               ProgressCapable,
+               Comparable<Activity>,
+               MotorDispenser,
+               NBConfigurable,
+               NBReconfigurable
+{
+    private static final Logger logger = LogManager.getLogger("ACTIVITY");
+
+    private final OpSequence<OpDispenser<? extends CycleOp<?>>> sequence;
+    private final ConcurrentHashMap<String, DriverAdapter<CycleOp<?>, Space>> adapters
+        = new ConcurrentHashMap<>();
+
+    public final ActivityMetrics metrics;
+    private ActivityMetricProgressMeter progressMeter;
+    private String workloadSource = "unspecified";
+    private RunState runState = RunState.Uninitialized;
+    private long startedAtMillis;
+    private final RunStateTally tally = new RunStateTally();
+    private ThreadLocal<RateLimiter> strideLimiterSource;
+    private ThreadLocal<RateLimiter> cycleLimiterSource;
+    private NBErrorHandler errorHandler;
+    private final List<AutoCloseable> closeables = new ArrayList<>();
+    private ErrorMetrics errorMetrics;
+    private Input input;
+    private StandardAction<?, ?> action;
+    private ActivityConfig config;
+
+    public Activity(NBComponent parent, ActivityConfig config) {
+        super(parent, NBLabels.forKV("activity", config.getAlias()).and(config.auxLabels()));
+        //        NBConfiguration validConfig = getConfigModel().apply(config.getMap());
+
+        this.applyConfig(config);
+        this.sequence = initSequence();
+        this.metrics = new ActivityMetrics(this);
+    }
+
+    public static ActivityConfig configFor(String s) {
+        return configFor(ParameterMap.parseParams(s).orElseThrow());
+    }
+
+
+    private OpSequence<OpDispenser<? extends CycleOp<?>>> initSequence() {
+        //        this.activityDef = activityDef;
+        //        this.metrics = new ActivityMetrics(this);
+
+        //        OpsDocList workload;
+        Optional<String> yaml_loc = config.getOptional("yaml", "workload");
+
+        // TODO: avoid having to load this duplicitously to parse the template variables in a separate phase
+        NBConfigModel yamlmodel = yaml_loc.map(path -> {
+            return OpsLoader.loadPath(path, new LinkedHashMap<>(config.getMap()), "activities")
+                .getConfigModel();
+        }).orElse(ConfigModel.of(Activity.class).asReadOnly());
```

Comment:
`yamlmodel` is unused. If you want to have the non-strict expansion of the TEMPLATE parameters into the Activity model then you will need to "reconfigure". This would be considered the difference between the now strict implementation and our current loose model.